pool/pool.go

```diff
@@ -222,13 +227,56 @@ func (p *Pool) StoreTx(ctx context.Context, tx types.Transaction, ip string, isW
 			log.Errorf("error adding event: %v", err)
 		}
 	}
+	var gasPrices GasPrices
+	// This is function get the gasPrice from memory (updated each 5s)
+	gasPrices.L1GasPrice, gasPrices.L2GasPrice = p.GetL1AndL2GasPrice()
+	err = p.ValidateBreakEvenGasPrice(ctx, tx, preExecutionResponse.txResponse.GasUsed, gasPrices)
+	if err != nil {
+		return err
+	}
 
 	poolTx := NewTransaction(tx, ip, isWIP)
 	poolTx.ZKCounters = preExecutionResponse.usedZkCounters
 
 	return p.storage.AddTx(ctx, *poolTx)
 }
 
+// ValidateBreakEvenGasPrice validates the effective gas price
+func (p *Pool) ValidateBreakEvenGasPrice(ctx context.Context, tx types.Transaction, preExecutionGasUsed uint64, gasPrices GasPrices) error {
+	breakEvenGasPrice, err := p.effectiveGasPrice.CalculateBreakEvenGasPrice(tx.Data(), tx.GasPrice(), preExecutionGasUsed, gasPrices.L1GasPrice)
+	if err != nil {
+		log.Errorf("egp-log: error calculating break even gas price: %v", err)
+		return err
+	}
+	margin := big.NewInt(0).SetUint64(p.cfg.EffectiveGasPrice.MarginBreakEven)
+	breakEvenGasPricePercentApplied := big.NewInt(0).Mul(breakEvenGasPrice, margin)
+	breakEvenGasPricePercentApplied = breakEvenGasPricePercentApplied.Div(breakEvenGasPricePercentApplied, big.NewInt(percentFactor))
+	log.Infof("egp-log:  breakEvenGasPrice: tx:%s breakEvenGasPrice:%v margin:%d breakEvenGasPricePercentApplied:%v L1GasPrice:%d L2GasPrice:%d tx.GasPrice():%v EGPenabled:%t",
+		tx.Hash().String(), breakEvenGasPrice,
+		p.cfg.EffectiveGasPrice.MarginBreakEven, breakEvenGasPricePercentApplied, gasPrices.L1GasPrice, gasPrices.L2GasPrice, tx.GasPrice(),
+		p.cfg.EffectiveGasPrice.Enabled)
+	if breakEvenGasPricePercentApplied.Cmp(tx.GasPrice()) == 1 { // breakEvenGasPricePercentApplied > tx.GasPrice()
+		// check against gasPrice now
+		L2GasPrice := big.NewInt(0).SetUint64(gasPrices.L2GasPrice)
+		if tx.GasPrice().Cmp(L2GasPrice) == -1 { // tx.GasPrice() < gasPrices.L2GasPrice
+			// Reject transaction if EffetiveGasPrice is enabled
+			if p.cfg.EffectiveGasPrice.Enabled {
+				// reject
+				log.Debugf("egp-log:  tx:%s Reject tx with gasPrice lower than L2GasPrice", tx.Hash().String())
+				return ErrEffectiveGasPriceGasPriceTooLow
+			}
+			log.Debug("egp-log:  tx:%s Accepted tx with gasPrice lower than L2GasPrice but EGP is disabled", tx.Hash().String())
+		} else {
+			// accept
+			log.Debugf("egp-log: egp-loss: tx:%s Accepted tx with loss because gasPrice in tx (%v) >= current L2GasPrice (%v)", tx.Hash().String(), tx.GasPrice(), L2GasPrice)
```

Comment:
I think this should be "egp-log" and log only one "egp-loss" (line 275). The idea is to have only one "egp-loss" for each tx since in this way it we will be more easy when filering logs looking for txs where we loss "money"